### DISCUSSION TITLE
Zip files download

### DISCUSSION THREAD

**Discussion by appliedai255** [jimmywarting/StreamSaver.js#204] [Q&A]
I have some zip files in a cloud server  (Size can be in GBs or TBs) that I want to download and track its download progress like how IDM does. StreanSaver.js will work for this?

**Comment by jimmywarting** [comment]
I don't know how IDM dose it... but yea, it could be possible to track progress with some middle pipeline that sits inbetween the download and save to disk stream that monitors the progress

but If you have a zip ready on the server then it's better to respond with content-disposition attachment header to trigger a download instead with regular http request instead of emulating it with service worker and javascript. it's always much better to trigger a background download instead of it happening on the worker+main thread. the isssue is that you must have the page open at all time... 

StreamSaver was intended for client side generated content first and formost.
